### PR TITLE
Move Native Library APIs to NativeLibrary class

### PR DESCRIPTION
**Pull request by swaroop-sridhar**
Move APIs that manipulate native libraries from
System.Runtime.InteropServices.Marshal class to the new
System.Runtime.InteropServices.NativeLibrary class.

In API review: #32015,
The LoadLibrary APIs were originally Approved to live in
System.Runtime.InteropServices.Marshal class.

However, recently the decision was changed such that the APIs are
in a new NativeLibrary class.

### CODE REVIEW

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        /// <summary>
+        /// NativeLibrary Loader: Simple API
+        /// This method is a wrapper around OS loader, using "default" flags.
+        /// </summary>
+        /// <param name="libraryPath">The name of the native library to be loaded</param>
+        /// <returns>The handle for the loaded native library</returns>  
+        /// <exception cref="System.ArgumentNullException">If libraryPath is null</exception>
+        /// <exception cref="System.DllNotFoundException ">If the library can't be found.</exception>
+        /// <exception cref="System.BadImageFormatException">If the library is not valid.</exception>
+        public static IntPtr LoadLibrary(string libraryPath)
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs]: Is `Library` still needed in the method names now that the `Library` is in the type name? "Library" in the 
method names looks superfluous:

```csharp
var handle = NativeLibrary.LoadLibrary("foo")
NativeLibrary.GetLibraryExport(handle, "m");
```

File: src/vm/nativelibrarynative.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// 
+// File: NativeLibraryNative.cpp
+//
+
+//
+// FCall's for the PInvoke classlibs
```
Comment (jkotas) [inline src/vm/nativelibrarynative.cpp]: Nit: This comment looks useless.

File: src/vm/nativelibrarynative.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// 
+// File: NativeLibraryNative.h
+//
+
+//
+// QCall's for the NativeLibrary class
```
Comment (AaronRobinsonMSFT) [inline src/vm/nativelibrarynative.h]: Superfluous.

Comment (swaroop-sridhar) [reply src/vm/nativelibrarynative.h]: I agre; but I followed the existing convention in other native.h files https://github.com/dotnet/coreclr/blob/master/src/vm/marshalnative.h#L9
https://github.com/dotnet/coreclr/blob/master/src/vm/assemblynative.hpp#L9

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+            return address != IntPtr.Zero;
+        }
+
+        /// External functions that implement the NativeLibrary interface
+
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        internal static extern IntPtr LoadLibraryFromPath(string libraryName, bool throwOnError);
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        internal static extern IntPtr LoadLibraryByName(string libraryName, RuntimeAssembly callingAssembly, 
+                                                        bool hasDllImportSearchPathFlag, uint dllImportSearchPathFlag,
```
Comment (AaronRobinsonMSFT) [inline src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs]: In this case, I think you want to specify the marshalling of the `bool`s. In this case, you are using C# `bool` which is 1 byte, but in the native signature below the Win32 `BOOL` value is being used which is 4-bytes. I can't recall the default off the top of my head, but I would suggest being explicit here - i.e. `[MarshalAs(UnmanagedType.Bool)]`

**Comment by swaroop-sridhar** [comment]
Thanks for the feedback @AaronRobinsonMSFT  and @jkotas 
I've addressed these comments in https://github.com/dotnet/coreclr/pull/21821/commits/b20b30e14e27f61aaf5692ffb71a42e00f664029

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs (hunk 4 above)
Comment (jkotas) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs]: FWIW, there are many QCalls with bool arguments - none of them have this extra ceremony.

Comment (swaroop-sridhar) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs]: OK https://github.com/dotnet/coreclr/pull/21821/commits/f1933b7b04ee4e09c7e335761f0b552bdce26ebd Revert the annotations, and squash to one commit.

**Comment by jkotas** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test please
@dotnet-bot test Ubuntu arm Cross Checked Innerloop Build and Test please

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs (hunk 4 above)
Comment (AaronRobinsonMSFT) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs]: Sounds good.

**Pull request merged**